diff --git a/usr/include/SDL2/SDL_config.h b/usr/include/SDL2/SDL_config.h
index aa9d35780..2698d5632 100644
--- ./usr/include/SDL2/SDL_config.h
+++ ./usr/include/SDL2/SDL_config.h
@@ -226,7 +226,7 @@
 #define HAVE_INOTIFY_INIT1 1
 #define HAVE_INOTIFY 1
 /* #undef HAVE_IBUS_IBUS_H */
-#define HAVE_IMMINTRIN_H 1
+/* #define HAVE_IMMINTRIN_H 1 */
 /* #undef HAVE_LIBUDEV_H */
 /* #undef HAVE_LIBSAMPLERATE_H */
 /* #undef HAVE_LIBDECOR_H */